File: lib/Erlang/primes.erl
-module(primes).
-export([primes_up_to/1]).

-import(lists, [any/2, reverse/1]).

primes_up_to(N) -> reverse(primes_up_to(N, 2, [])).

primes_up_to(N, I, Primes) when I > N -> Primes;
primes_up_to(N, I, Primes) ->
  case is_prime(I, Primes) of
    true -> primes_up_to(N, I+1, [I | Primes]);
    false -> primes_up_to(N, I+1, Primes)
  end.
  
is_prime(I, Primes) -> not any(fun(P) -> I rem P == 0 end, Primes).

File: lib/Python/primes.py
from functional import foldl

def primes_up_to(n):
    return primes_up_to_acc(n, 2, [])
    
def primes_up_to_acc(n, i, primes):
    if i > n:
        return primes
    else:
        if is_prime(i, primes):
            primes.append(i)
        primes_up_to_acc(n, i+1, primes)
        return primes

def is_prime(i, primes):
    return not foldl(lambda a, b: a or b, False, [i % p == 0 for p in primes])
